lib/zip/output_stream.rb

```diff
@@ -95,14 +95,14 @@ def put_next_entry(entry_name, comment = nil, extra = nil, compression_method =
       new_entry = if entry_name.kind_of?(Entry)
                     entry_name
                   else
-                    Entry.new(@file_name, entry_name.to_s)
+                    e = Entry.new(@file_name, entry_name.to_s, comment,
+                                  compression_method, level)
+                    e.extra = extra
```

Comment:
Previously this assignment would only happen if `extra` was non-`nil`. It looks like everything still goes through OK with the new setter, but it relies `nil.kind_of?` and `nil.to_s`, which is fine but a bit tricky, so I thought I should check whether this was the intention.

Reply:
Erk. Good catch, thanks. I've changed the signature of `OutputStream#put_next_entry` so that `extra` doesn't default to `nil`, **and** I've made the setter more robust.